tldw_Server_API/app/core/DB_Management/ChaChaNotes_DB.py

```
@@ -9359,6 +9677,359 @@ def _check_answer(self, question: Dict[str, Any], user_answer: Any) -> bool:
+            expected_version,
+            allowed_fields,
+            pk_col_name="id",
+        )
+
+    def soft_delete_writing_session(self, session_id: str, expected_version: int) -> bool | None:
+        return self._soft_delete_generic_item("writing_sessions", session_id, expected_version, pk_col_name="id")
+
+    def clone_writing_session(self, session_id: str, *, name: Optional[str] = None) -> Dict[str, Any]:
+        existing = self.get_writing_session(session_id)
+        if not existing:
+            raise ConflictError("Writing session not found.", entity="writing_sessions", entity_id=session_id)
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)